test/fixtures/sem-html5-scenarios/paragraph-with-title.html

```diff
@@ -0,0 +1,4 @@
+<p class="paragraph">
+<strong class="title">Famous intro</strong>
```

Comment:
Since permitted content in a paragraph are [phrasing content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content), we could use `<label>` instead of the `<strong>` tag:

```html
<p class="paragraph">
<label class="title">Famous intro</label >
A long time ago in a galaxy far, far away.
</p>
```

> The HTML <label> element represents a caption for an item in a user interface.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

Reply:
For reference, we cannot use heading tag because it's not permitted in a paragraph.

Reply:
Putting the title/caption inside the p is definitely the approach we want to take as it removes the complication (and bad taste) introduced by enclosing the p tag.

I agree the heading tag won't work here. But I don't think label is the right choice either. That's really reserved for a form, so we would be diluting its meaning. I think the safe choice here is either span or strong. The advantage of strong is that it comes prestyled.

What Asciidoctor EPUB3 uses is `<strong class="head">`. Though if "head" seems too obtuse, I would be supportive of the class name "title".

Reply:
> But I don't think label is the right choice either.

I came to the same conclusion in another issue. I don't think we should use `<label>`.

Reply:
> Though if "head" seems too obtuse, I would be supportive of the class name "title".

I can see the reason for that naming. We could also use `lead` since the title is now inside the paragraph and "leading" the text but title is probably less disruptive and more obvious.

Reply:
In publishing, the formal name for this construct is a "run-in heading". See https://docs.microsoft.com/en-us/style-guide/scannable-content/headings#using-run-in-headings and http://www.creativeglossary.com/graphic-design/run-in-heading.html. That's how I ended up with "head" in Asciidoctor EPUB3. "lead" actually has a very specific meaning in publishing and not something we can use here.

Reply:
The DocBook stylesheets use the strong tag for the run-in heading. I really think that's the best choice.

We may also consider adding the ability to control the "stop" character that follows the run-in heading. Though that certainly should be a separate issue.

Reply:
Late to the discussion, but isn't it a significant change to the semantics to change a paragraph title to a run in?

Reply:
I would argue it's always been a run-in heading. That's what the DocBook stylesheets produce. So technically, it was the built-in converter that was doing it incorrectly. We've also now done it this way in Asciidoctor EPUB3 for many years, so it's something that is very common already.

Reply:
Fair enough, except its now just a piece of bolded text, its actually lost its title semantic.  That means for example a screen reader won't identify it as a title.

Reply:
My CSS is pretty poor but `run-in` is a CSSable effect IIUC and it can be applied to an element before the `<p>` so that element could retain `<header>` semantics?

Reply:
I don't think we should rely on `run-in`. MDN states that this feature is experimental (and should not be used in production code): https://developer.mozilla.org/en-US/docs/Web/CSS/display

The specification is also not very encouraging:

> The following features are at-risk, and may be dropped during the CR period:
> ...
> display: run-in

https://drafts.csswg.org/css-display/#status

If we want to add a relation between the title and the paragraph we could use [`aria-describedb`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-describedby_attribute) / [`aria-labelledby`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-labelledby_attribute) but we would need to generate an id.

>

We can add `role=heading`

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/heading_role
I don't own a screen reader, so I'm not sure what are our alternatives if we don't want or cannot use a `<h6>`.


For reference, here's what we produce today:

```html
<div class="paragraph">
  <div class="title">Famous intro</div>
  <p>A long time ago in a galaxy far, far away.</p>
</div>
```

Reply:
> I don't think we should rely on run-in. MDN states that this feature is experimental
> Could you elaborate on your proposal and post the HTML it would produce?

See I told you my CSS was poor, and my HTML is not much better :grin:

Really what I'm trying to do is to jog you experts that the semantics of title is being lost if its converted to "just bolded text".  

Related I guess is: what does a title on other blocks like delimited ones translate to in semantic HTML?

> For reference, here's what we produce today:

But of course thats a non-semantic converter :grinning:

Reply:
> Related I guess is: what does a title on other blocks like delimited ones translate to in semantic HTML?

It depends, I believe that we are going to use [`<figcaption>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption) on images (and potentially other blocks) and [`<caption>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/caption) on tables, but a lot is still undecided...
Having said that, if your question is "do we want to use the same semantic structure on all blocks?" then the answer is probably no.